src/JsonSchema/Constraints/Constraint.php

```diff
@@ -33,37 +37,54 @@
     private $factory;
 
     /**
-     * @param int          $checkMode
-     * @param UriRetriever $uriRetriever
-     * @param Factory      $factory
+     * @param int           $checkMode
+     * @param SchemaStorage $schemaStorage
+     * @param UriRetriever  $uriRetriever
+     * @param Factory       $factory
      */
-    public function __construct($checkMode = self::CHECK_MODE_NORMAL, UriRetriever $uriRetriever = null, Factory $factory = null)
-    {
-        $this->checkMode    = $checkMode;
-        $this->uriRetriever = $uriRetriever;
-        $this->factory = $factory;
+    public function __construct(
+        $checkMode = self::CHECK_MODE_NORMAL,
+        SchemaStorage $schemaStorage = null,
+        UriRetriever $uriRetriever = null,
+        Factory $factory = null
+    ) {
```

Comment:
wouldn't it be kinda easier to move `$schemaStorage` to the end of the list? if we'd append it to the existing list, we wouldn't introduce a BC-Break at this point.